modules/server/src/main/scala/navigate/server/NavigateEngine.scala

```diff
@@ -248,17 +250,34 @@ object NavigateEngine {
       )
     )
 
-    override def slew(slewOptions: SlewOptions, tcsConfig: TcsConfig): F[Unit] = command(
-      engine,
-      Slew,
-      transformCommand(
-        Slew,
-        Handler.modify[F, State, ApplyCommandResult](_.focus(_.onSwappedTarget).replace(false)) *>
-          Handler.fromStream(
-            Stream.eval(systems.tcsCommon.slew(slewOptions, tcsConfig))
+    override def slew(
+      slewOptions: SlewOptions,
+      tcsConfig:   TcsConfig,
+      oid:         Option[Observation.Id]
+    ): F[Unit] =
+      Logger[F].info(s"Starting slew to ${oid}") *>
+        command(
+          engine,
+          Slew,
+          transformCommand(
+            Slew,
+            Handler.modify[F, State, ApplyCommandResult](
+              _.focus(_.onSwappedTarget).replace(false)
+            ) *>
+              Handler.fromStream(
+                Stream.eval(
+                  systems.tcsCommon
+                    .slew(slewOptions, tcsConfig)
+                    // if succesful send an event to the odb
+                    .flatTap(_ =>
+                      oid
+                        .map(systems.odb.addSlewEvent(_, SlewStage.StartSlew))
+                        .getOrElse(Applicative[F].unit)
+                    )
+                )
```

Comment:
Here we can add the EndSlew after a wait for the in-position flag, either inside the `eval`, or as the next element in the Stream (that way we can produce extra events to inform the ui)

Reply:
After some discussions, It seems we don't need an end-slue event after all.